### PR TITLE
OCPBUGS-43454: Ignore cluster manager `topology not managed` errors for localnet  with no subnets

### PR DESCRIPTION
**Pull request by maiqueb**
#### What this PR does and why is it needed

This PR ignores `topology net managed` errors on the initial all networks sync of the cluster manager controllers.

Only the `secondaryNetworkClusterManager` - which is created by the cluster controller manager, and executed in the control plane pods - actually returns this type of error.

#### Which issue(s) this PR fixes

Fixes #

#### How to verify it

1. Configure the bridge mappings for the localnet network
2. Create a localnet network without subnets.
3. Attach a pod to the localnet network.
4. Restart the ovnkube-control-plane pods

To configure the mappings for the localnet network you can (upstream):
```sh
for pod in $(kubectl get pods -novn-kubernetes -l "app=ovs-node" --no-headers -ocustom-columns=":metadata.name"); do 
    kubectl -n ovn-kubernetes exec $pod -- ovs-vsctl set open . external-ids:ovn-bridge-mappings="physnet:breth0,tenantblue:breth0"
done
```

The following manifests can be used to reproduce this upstream:
```yaml
---
apiVersion: k8s.cni.cncf.io/v1
kind: NetworkAttachmentDefinition
metadata:
  name: tenantblue
spec:
    config: '{
        "cniVersion": "0.4.0",
        "name": "tenantblue",
        "netAttachDefName": "default/tenantblue",
        "topology": "localnet",
        "type": "ovn-k8s-cni-overlay",
        "logFile": "/var/log/ovn-kubernetes/ovn-k8s-cni-overlay.log",
        "logLevel": "5",
        "logfile-maxsize": 100,
        "logfile-maxbackups": 5,
        "logfile-maxage": 5
    }'
---
apiVersion: v1                
kind: Pod                        
metadata:
  namespace: default
  name: pod1
  annotations:
    k8s.v1.cni.cncf.io/networks: '[
      {
        "name": "tenantblue","ips": ["192.168.200.20/24"]
      }                                          
    ]'                         
spec:                                 
  containers:                                                        
  - name: centos        
    image: docker.io/centos/tools:latest
    command:                    
    - /sbin/init           
```

Without this PR, the control-plane pods will crash loop.

#### Description for the changelog

Prevent a localnet network without subnets (typical virt use case) from crash-looping the control plane pods when they restart (for instance, during an upgrade).

#### Does this PR introduce a user-facing change?

```release-note
Prevent a localnet network without subnets from crash-looping the control plane pods upon restart (e.g. during an upgrade)
```

### CODE REVIEW

File: go-controller/pkg/network-attach-def-controller/network_manager.go [hunk 1]
```diff
@@ -214,7 +215,13 @@ func (nm *networkManagerImpl) syncAll() error {
 	start := time.Now()
 	klog.Infof("%s: syncing all networks", nm.name)
 	for _, network := range validNetworks {
-		if err := nm.sync(network.GetNetworkName()); err != nil {
+		if err := nm.sync(network.GetNetworkName()); errors.Is(err, ErrNetworkControllerTopologyNotManaged) {
```
Comment (tssurya) [inline go-controller/pkg/network-attach-def-controller/network_manager.go:218]: do we have a unit test for localnet with no subnets?

**Comment by tssurya** [comment]
do we have a unit test for localnet with no subnets?

File: go-controller/pkg/network-attach-def-controller/network_manager.go (hunk 1 above)
Comment (tssurya) [inline go-controller/pkg/network-attach-def-controller/network_manager.go:218]: wow learnt something new here, so from code perspective, its a localnet network of type ovn-k8s-cni-overlay with  no subnets meaning its pure l2 usecase ? OR something else does IPAM for those VM pods and they talk to the VLAN outside?

so this means we don't create any NAD controller for such networks? hence ErrNetworkControllerTopologyNotManaged ?

Comment (maiqueb) [reply go-controller/pkg/network-attach-def-controller/network_manager.go:218]: No, we have one, but running in the ovnkube-node.

**Comment by maiqueb** [comment]
> do we have a unit test for localnet with no subnets?

I can add a test case that ensure this error is thrown for this network, but I'm not sure if I can add more than that.

File: go-controller/pkg/network-attach-def-controller/network_manager.go (hunk 1 above)
Comment (tssurya) [reply go-controller/pkg/network-attach-def-controller/network_manager.go:218]: Hmm why do we create one in the node side? Was it for dpu use case?

Comment (tssurya) [reply go-controller/pkg/network-attach-def-controller/network_manager.go:218]: what does that controller do on the node side? is it for the DPU case?

**Review by jcaamano — APPROVED** [review]

**Pull request merged**